client/src/components/Meeting/Meeting.tsx

```diff
@@ -1,66 +1,112 @@
+};
+
 const Meetings = ({ loggedInUser }: Props): JSX.Element => {
+  const { updateSnackBarMessage } = useSnackBar();
   const classes = useStyles();
-  const [spacing] = React.useState<GridSpacing>(2);
+  const [events, setEvents] = useState([]);
+
+  useEffect(() => {
+    fetchEvents().then((data) => {
+      if (data.success) {
+        setEvents(data.success);
+      } else {
+        updateSnackBarMessage('Could not recieve events from backend server');
+      }
+    });
+    // eslint-disable-next-line
+  }, []);
 
+  const copyToClipBoard = (url: string) => {
+    navigator.clipboard.writeText(url);
+  };
   return (
-    <Grid container spacing={3}>
-      <Grid item xs={12}>
-        <Divider />
-        <AvatarDisplay loggedIn user={loggedInUser} />
-        <Typography variant="h5">{loggedInUser.username}</Typography>
-        <Typography variant="h5">{loggedInUser.email}</Typography>
-        <Button variant="outlined" color="primary" href="#outlined-buttons" className={classes.right}>
-          <Typography variant="h6">
+    <Box>
+      <Grid container xs={12} direction="row" className={classes.profileSection}>
+        <Box display="flex">
+          <Box>
+            <AvatarDisplay loggedIn user={loggedInUser} />
+          </Box>
+          <Box marginLeft={2}>
+            <Typography component="h3" className={classes.title}>
+              {loggedInUser.username}
+            </Typography>
+            <Typography variant="subtitle1">{loggedInUser.email}</Typography>
+          </Box>
+        </Box>
+        <Box>
+          <Button variant="outlined" color="primary" className={classes.newEventButton}>
             <AddIcon />
-            New Event
-          </Typography>
-        </Button>
+            <Typography variant="h6">New Event</Typography>
+          </Button>
+        </Box>
       </Grid>
-      <Grid item xs={12}>
-        <Grid container direction="row" alignItems="center" justify="center" spacing={spacing}>
-          {[0, 1, 2].map((value) => (
-            <Grid key={value} item xs={3}>
-              <Paper className={classes.paper}>
-                <Typography variant="h5">15 min Meeting</Typography>
-                <SettingsIcon />
-                <Divider />
-                <ScheduleIcon />
-                <Typography variant="h5">15</Typography>
-                <Button variant="outlined" color="primary">
-                  <Link to={`/${loggedInUser.username}/15min`}>Create Meeting</Link>
-                </Button>
-              </Paper>
-            </Grid>
-          ))}
-        </Grid>
-        <Button variant="outlined" color="primary" href="#outlined-buttons" className={classes.right}>
-          Getting Started!
-        </Button>
+      <Divider className={classes.divider} />
+      <Grid container xs={12} className={classes.meetingContainer} spacing={3}>
+        {events?.map((eachEvent: any, index: any) => (
+          <Grid key={index} item xs={12} md={4}>
```

Comment:
The React contributors do not recommend using indexes as keys because it can lead to bad component behaviour and performance issues. Is there something more unique we could utilize for the key.